src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -13,11 +11,10 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
-    internal sealed class LambdaSymbol : SourceMethodSymbol
+    internal sealed class LambdaSymbol : SourceMethodSymbolWithAttributes
```

Comment:
>LambdaSymbol [](start = 26, length = 12)

It looks like this type should override AddDeclarationDiagnostics so that it could collect errors that are reported during attribute binding and other similar errors that are normally treated as declaratio errors.